### PR TITLE
[NDTensors][ITensorGPU] Debug CUDA and upkeep

### PR DESCRIPTION
**Pull request by kmp5VT**
Branch to remove `to_vector_type` fix some of the default typing problems and also fix the CUDA runtime problem without restricting CUDA version

### CODE REVIEW

File: ITensorGPU/test/Project.toml [hunk 1]
```diff
@@ -5,3 +5,7 @@ ITensors = "9136182c-28ba-11e9-034c-db9fb085ebd5"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
```
Comment (mtfishman) [inline ITensorGPU/test/Project.toml:8]: Do we need `Pkg` as a test dependency? I don't see it used anywhere.

Comment (kmp5VT) [reply ITensorGPU/test/Project.toml:8]: I don't think it is necessary

**Comment by mtfishman** [comment]
Looks good, thanks!

File: jenkins/Jenkinsfile [hunk 2]
```diff
@@ -27,7 +27,7 @@ pipeline {
           }
           steps {
             sh '''
-              julia --project=monorepo -e 'using Pkg; Pkg.Registry.update(); Pkg.develop(; path="./NDTensors"); Pkg.develop(; path="."); Pkg.develop(; path="./ITensorGPU"); Pkg.test("ITensorGPU");'
+              julia -e 'using Pkg; Pkg.Registry.update(); Pkg.develop(path="./NDTensors"); Pkg.develop(; path="./ITensorGPU"); Pkg.add("CUDA"); using CUDA; CUDA.set_runtime_version!("v11.8"); Pkg.test("ITensorGPU");'
```
Comment (mtfishman) [inline jenkins/Jenkinsfile:30]: This new version seems to drop testing against the development version of `ITensors` (the operation `Pkg.develop(; path=".")`), I think we should keep that.

Comment (kmp5VT) [reply jenkins/Jenkinsfile]: Okay, I added it back!

**Pull request merged**